keras_cv/models/task.py

```diff
@@ -112,57 +154,29 @@ def from_preset(
         ```
         """
 
-        if not cls.presets:
-            raise NotImplementedError(
-                "No presets have been created for this class."
+        # TODO: delete me!
+        if preset in cls.presets:
+            return cls._legacy_from_preset(
+                preset, load_weights, input_shape, **kwargs
             )
 
-        if preset not in cls.presets:
-            raise ValueError(
-                "`preset` must be one of "
-                f"""{", ".join(cls.presets)}. Received: {preset}."""
-            )
+        check_preset_class(preset, cls)
 
-        if load_weights and preset not in cls.presets_with_weights:
-            raise ValueError(
-                f"""Pretrained weights not available for preset "{preset}". """
-                "Set `load_weights=False` to use this preset or choose one of "
-                "the following presets with weights:"
-                f""" "{'", "'.join(cls.presets_with_weights)}"."""
-            )
-
-        metadata = cls.presets[preset]
-        # Check if preset is backbone-only model
+        # Backbone case.
         if preset in cls.backbone_presets:
```

Comment:
This we need to reformulate. The goal is to be able to load a "preset directory" from disk, kaggle, etc, canonically, without needing metadata about it that preset in particular in the library itself (`cls.backbone_presets`). See https://github.com/keras-team/keras-nlp/blob/kaggle/keras_nlp/models/task.py#L219, the difference is we are just checking the preset config against a fixed class to determine if we are in the `backbone` or `task` case.

Here, since KerasCV allows mixing and matching backbones, we probably want to check if the preset class is a subclass of `Backbone`?

Reply:
For simplicity sake, I've reformulated it similar to the KerasNLP approach. I've added a property for `backbone_cls` in Task and subclassed it where applicable (only 2 instances). Thanks for catching this!

Reply:
Ah missed this comment and replied elsewhere. But I think that while a 1-1 mapping of backbone cls <> task cls is a valid assumption for KerasNLP, is it not for KerasCV, so we cannot do this here. We need to allow mixing and matching of backbones with a task.